app/code/Magento/PageCache/Observer/FlushFormKey.php

```diff
@@ -38,7 +46,14 @@ public function __construct(CookieFormKey $cookieFormKey, DataFormKey $dataFormK
      */
     public function execute(\Magento\Framework\Event\Observer $observer)
     {
+        $currentFormKey = $this->dataFormKey->getFormKey();
         $this->cookieFormKey->delete();
         $this->dataFormKey->set(null);
+
+        $beforeParams = $this->customerSession->getBeforeRequestParams();
+        if ($beforeParams['form_key'] == $currentFormKey) {
```

Comment:
Could you add check that $beforeParams['form_key'] is exist? $beforeParams might be empty array.
Also would be great to use strict equal === comparison instead of ==.

Reply:
Thanks @ihor-sviziev I will work after mm17es in this fix.